kubernetes/monitoring/services/mongo/monit-mongo.yaml

```diff
@@ -0,0 +1,164 @@
+# [ConfigMap] [Service] [StatefulSet]
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: monit-mongo-configmap
+  namespace: monit-mongo
+data:
+  mongo.conf: |
+    storage:
+      dbPath: /data/db
+  ensure-users.js: |
+    const targetDbStr = 'test';
+    const rootUser = process.env["$MONGO_ROOT_USERNAME"];
+    const rootPass = process.env["MONGO_ROOT_PASSWORD"];
+    const usersStr = process.env["MONGO_USERS_LIST"];
+
+    const adminDb = db.getSiblingDB('admin');
+    adminDb.auth(rootUser, rootPass);
+    print('Successfully authenticated admin user');
+
+    const targetDb = db.getSiblingDB(targetDbStr);
+
+    const customRoles = adminDb
+      .getRoles({rolesInfo: 1, showBuiltinRoles: false})
+      .map(role => role.role)
+      .filter(Boolean);
+
+    usersStr
+      .trim()
+      .split(';')
+      .map(s => s.split(':'))
+      .forEach(user => {
+        const username = user[0];
+        const rolesStr = user[1];
+        const password = user[2];
+
+        if (!rolesStr || !password) {
+          return;
+        }
+
+        const roles = rolesStr.split(',');
+        const userDoc = {
+          user: username,
+          pwd: password,
+        };
+
+        userDoc.roles = roles.map(role => {
+          if (!~customRoles.indexOf(role)) {
+            return role;
+          }
+          return {role: role, db: 'admin'}; 
+        });
+
+        try {
+          targetDb.createUser(userDoc);
+        } catch (err) {
+          if (!~err.message.toLowerCase().indexOf('duplicate')) {
+            throw err;
+          }
+        }
+      });
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: monit-mongo
+  namespace: monit-mongo
+spec:
+  type: NodePort
+  ports:
+    - port: 27017
+      protocol: TCP
+      targetPort: 27017
+      nodePort: 32000
```

Comment:
once again, why do you need nodePort? k8s provide excellent capabilities to reach services and we rarely need to expose service ports. Instead, you can reach your service as simple as `<svc-name>.<namesapce>.svc.cluster.local:<port>`. For instance, in auth proxy server, I use it all the time to redirect HTTP request to, e.g. `dbs2go.dbs.svc.cluster.local:8217`. I suggest that you evaluate the needs for nodePort and make proper adjustments.